src/storage.js

```diff
@@ -15,16 +15,23 @@ class Storage {
         sessionstorage.setItem(key, obj);
     }
 
-    static getJSON(key: string): ?(Object|Array) {
+    static getJSON(key: string): any {
+        var item = this.getItem(key);
+
+        if (typeof item != "string") {
+            console.error("Cannot getJSON with non-string ");
+            return null;
+        }
+
         try {
-            return JSON.parse(this.getItem(key));
+            return JSON.parse(item);
         } catch (error) {
             console.error(error);
             return null;
         }
     }
 
-    static setJSON(key: string, obj: Object|Array): void {
+    static setJSON(key: string, obj: mixed): void {
```

Comment:
I don't think that's correct. According to the docs:

    Sometimes, multiple signatures are not needed to express overloading: the signatures can 
    be coalesced using union types. Flow provides the following syntax for union types:

    T1 | .. | Tn

    is the union of types Ti. Union types are available for general use.